### PR TITLE
Fix the status of sticky fields after syncing with AnkiWeb

### LINKED ISSUE #12770: [Bug] Add note window doesn't indicate which fields are pinned when it 'remembers' the pinned fields from Desktop Anki
###### Reproduction Steps
1. Select add note option on Desktop Anki.
2. Pin one or more fields and create some notes.
3. Sync collection with AnkiDroid.
4. Open Add note window on AnkiDroid.

###### Expected Result
1. AnkiDroid (and Desktop Anki) remembers which fields were pinned.
2. The add note window indicates which fields are pinned (as it does when you click on the pin option).

###### Actual Result
Desktop Anki remembers as well as indicates the pinned fields.
However, AnkiDroid just remembers but does not indicate the pinned fields.

###### Debug info
AnkiDroid Version = 2.16alpha89

Android Version = 7.0

Manufacturer = motorola

Model = Moto C Plus

Hardware = mt6735

Webview User Agent = Mozilla/5.0 (Linux; Android 7.0; Moto C Plus Build/NRD90M.03.045; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/107.0.5304.91 Mobile Safari/537.36

ACRA UUID = c6a580cb-02cc-4970-b7a5-24615e6542e1

New schema = true

Scheduler = std3

Crash Reports Enabled = true

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by error96num**
## Fixes
Fixes #12770
This was caused by the inconsistency between actual field state and sticky button state.

## Approach
Put the synced sticky field's index to the map's key before registering a click listener for the `toggleStickyButton`.

## How Has This Been Tested?
Mobile
- Google Pixel 6 Pro (physical device)
- Android 13

Desktop (used for sync)
- MacBook Pro
- Apple M1 Pro
- macOS Monterey 12.4
- ⁨Anki 2.1.54

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -1411,6 +1411,8 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
     }
 
     private fun setToggleStickyButtonListener(toggleStickyButton: ImageButton?, index: Int) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1413]: Thanks!

Could you add a `@NeedsTest("explanation")` so we can go through this later and add a regression test?

**Comment by david-allison** [comment]
Thanks!

**Review by mikehardy — APPROVED** [review]
Looks good to me and I assume based on David's actions (adding "needs second review" label) that he actually approves (hope that's correct!)

I altered the description to contain the github-specific `Fixes #NNNN` annotation so that it is completely linked with / auto-closes the related issue

Thanks!

**Comment by david-allison** [comment]
To confirm: That was an approval

**Pull request merged**